### ISSUE TITLE
NGXLoggerServerService - sendToServer sends original message instead of secured one

### ISSUE THREAD

**Issue by fbo3264** [dbfannin/ngx-logger#286]
In the `sendToServer` method of `NGXLoggerServerService` a copy of the metadata is created and any modifications by `secureAdditionalParameter` or `secureMessage` are performed on the copy, which is fine of course. However, afterwards when determining the `requestBody` the original `metadata`-object is used, which means all modifications of `localMetadata.message` get ignored. If - for example - an error-object is logged, then `localMetadata.message` will contain only the stackTrace-string, but metadata will still contain the error object. Since `requestBody` is later passed to `logOnServer` an empty object `{}` will be sent instead of the error-stacktrace.

```
public sendToServer(metadata: INGXLoggerMetadata, config: INGXLoggerConfig): void {
    // Copying metadata locally because we don't want to change the object for the caller
    const localMetadata = { ...metadata };

    localMetadata.additional = this.secureAdditionalParameters(localMetadata.additional);

    localMetadata.message = this.secureMessage(metadata.message);

    // Allow users to customise the data sent to the API
    const requestBody = this.customiseRequestBody(metadata);

   // ...

   this.logOnServer<any>(config.serverLoggingUrl, requestBody, { ... });
```

In my opinion it should be something like:

```
public sendToServer(metadata: INGXLoggerMetadata, config: INGXLoggerConfig): void {
    // Copying metadata locally because we don't want to change the object for the caller
    const localMetadata = { ...metadata };

    localMetadata.additional = this.secureAdditionalParameters(localMetadata.additional);

    localMetadata.message = this.secureMessage(localMetadata.message);

    // Allow users to customise the data sent to the API
    const requestBody = this.customiseRequestBody(localMetadata);

   // ...

   this.logOnServer<any>(config.serverLoggingUrl, requestBody, { ... });
```

**Comment by bmtheo** [comment]
Yes indeed ! thanks for this feedback

I'll fix this in the weekend

**Comment by bmtheo** [comment]
Fix available in 5.0.7 :)

**Comment by fbo3264** [comment]
Thanks a lot, works perfectly now! :)
